lib/delayed/priority.rb

```diff
@@ -98,13 +114,20 @@ def default_alerts
         @names ? {} : DEFAULT_ALERTS
       end
 
+      def names_to_midpoint_priority
+        names.each_cons(2).to_h { |(name, priority_value), (_, next_priority_value)|
+          midpoint = priority_value.to_i + ((next_priority_value.to_i - priority_value.to_i).to_d / 2).ceil
```

Comment:
I think we could avoid the `to_i` calls by adding the math operator functions for addition & subtraction (similar to the custom comparison operator), but wanted to get thoughts on that first